lua/tardis/screens/sh_screenbutton.lua

```diff
@@ -0,0 +1,282 @@
+	then
+		self:SetImages(file_off)
+	else
+		self.label:SetColor(Color(0,0,0,255))
+		self.label:SetText("   "..text)
+	end
+end
+
+function TardisScreenButton:SetFont(font)
+	self.label:SetFont(font)
+end
+
+function TardisScreenButton:GetPosX()
+	return self.pos[1]
+end
+function TardisScreenButton:GetPosY()
+	return self.pos[2]
+end
+function TardisScreenButton:GetWide()
+	return self.size[1]
+end
+function TardisScreenButton:GetTall()
+	return self.size[2]
+end
+function TardisScreenButton:GetSize()
+	return self.size[1], self.size[2]
+end
+
+function TardisScreenButton:SetImages(off, on)
+	self.icon_off = off
+	self.icon_on = on or off
+	self.icon:SetImage(self.icon_off)
+end
+
+function TardisScreenButton:SetIsToggle(is_toggle)
+	self.is_toggle=is_toggle
+end
+
+function TardisScreenButton:SetPressed(on)
+	self.on = on
+	if on then
+		self.icon:SetImage(self.icon_on)
+	else
+		self.icon:SetImage(self.icon_off)
+	end
+end
+function TardisScreenButton:IsPressed()
+	return self.on
+end
+
+function TardisScreenButton:SetVisible(visible)
+	self.visible = visible
+	self.icon:SetVisible(visible)
+	self.label:SetVisible(visible)
+end
+
+function TardisScreenButton:IsVisible()
+	return self.visible
+end
+
+function TardisScreenButton:SetTransparency(x)
+	self.transparency = x
+end
+
+function TardisScreenButton:SetControl(control)
+	self.DoClick = function()
+		TARDIS:Control(control)
+	end
+end
+
+function TardisScreenButton:SetPressedStateData(ext, data1, data2)
+	if data2 == nil
+	then
+		self.Think = function()
+			self:SetPressed(ext:GetData(data1))
+		end
+	else
+		self.Think = function()
+			self:SetPressed(ext:GetData(data1) or ext:GetData(data2))
+		end
+	end
+end
+
+function TardisScreenButton:InitiateMove(x, y, relative, speed)
+	if self.moving.now then return end
+
+	local moving = {}
+	moving.now = true
+	moving.parent = self
+	moving.last = CurTime()
+	moving.speed = speed or 100
+
+	if relative
+	then
+		moving.aim = { self.pos[1] + x, self.pos[2] + y }
+		moving.step = { x, y }
+	else
+		moving.aim = { x, y }
+		moving.step = { x - self.pos[1], y - self.pos[2] }
+	end
+
+	moving.now_outside = (self.pos[1] < 0) or (self.pos[2] < 0)
+		or (self.pos[1] + self.size[1] > self.parent:GetWide())
+		or (self.pos[2] + self.size[2] > self.parent:GetTall())
+
+	moving.aim_outside = (moving.aim[1] < 0) or (moving.aim[2] < 0)
+		or (moving.aim[1] + self.size[1] > self.parent:GetWide())
+		or (moving.aim[2] + self.size[2] > self.parent:GetTall())
+
+
+	moving.step = { moving.step[1] * moving.speed / 1000, moving.step[2] * moving.speed / 1000 }
+	moving.transp_step = 0
+	if moving.now_outside and moving.aim_outside
+	then
+		self.transparency = 0
+		moving.transp_step = 0
+	elseif moving.now_outside
+	then
+		self.transparency = 0
+		moving.transp_step = 255 * speed / 1000;
+	elseif moving.aim_outside
+	then
+		self.transparency = 255
+		moving.transp_step = - 255 * speed / 1000;
+	end
+
+	moving.move = function()
```

Comment:
The speed of this move is dependent on framerate, use FrameTime() to compensate for this

Reply:
Suggested all changes, please review and add

Reply:
Should be fixed and work normally since e04807cdfdbea3965c9258753dc9a3192c01048a
Please, test/review and resolve conversation if it's okay

Reply:
Fixed now